### PR TITLE
[#3920] feat(catalog-lakehouse-paimon): Support jdbc backend for Paimon Catalog

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?

Support jdbc backend for Paimon Catalog

### Why are the changes needed?

Fix: https://github.com/apache/gravitino/issues/3920

### Does this PR introduce _any_ user-facing change?
yes, updated the related doc.

### How was this patch tested?
new UT and IT.

### CODE REVIEW

**Comment by caican00** [comment]
Hi @FANNG1 could you help review this pr? Thanks!

File: catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java [hunk 1]
```diff
@@ -46,14 +46,26 @@ public class PaimonCatalogPropertiesMetadata extends BaseCatalogPropertiesMetada
   public static final String PAIMON_METASTORE = "metastore";
   public static final String WAREHOUSE = "warehouse";
   public static final String URI = "uri";
+  public static final String JDBC_USER = "jdbc.user";
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:49]: could you use `jdbc-user` like Iceberg?

File: catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/utils/CatalogUtils.java [hunk 2]
```diff
@@ -121,6 +123,17 @@ private static void checkPaimonConfig(PaimonConfig paimonConfig) {
       Preconditions.checkArgument(
           StringUtils.isNotBlank(uri), "Paimon Catalog uri can not be null or empty.");
     }
+
+    if (PaimonCatalogBackend.JDBC.name().equalsIgnoreCase(metastore)) {
+      String jdbcUser = paimonConfig.get(CATALOG_JDBC_USER);
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/utils/CatalogUtils.java:128]: you could add validate check in ConfigEntry like 
```java
  public static final ConfigEntry<String> JDBC_USER =
      new ConfigBuilder(IcebergConstants.GRAVITINO_JDBC_USER)
          .doc("The username of the Jdbc connection")
          .version(ConfigConstants.VERSION_0_2_0)
          .stringConf()
          .checkValue(StringUtils::isNotBlank, ConfigConstants.NOT_BLANK_ERROR_MSG)
          .create();
```

File: catalogs/catalog-lakehouse-paimon/build.gradle.kts [hunk 3]
```diff
@@ -122,12 +122,14 @@ dependencies {
...
   testImplementation(libs.mysql.driver)
   testImplementation(libs.postgresql.driver)
+  testImplementation(libs.h2db)
   testImplementation(libs.bundles.log4j)
   testImplementation(libs.junit.jupiter.params)
   testImplementation(libs.paimon.s3)
   testImplementation(libs.paimon.spark)
   testImplementation(libs.testcontainers)
   testImplementation(libs.testcontainers.localstack)
+  testImplementation(libs.testcontainers.mysql)
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-paimon/build.gradle.kts:132]: why adding this?

File: catalogs/catalog-lakehouse-paimon/src/test/java/org/apache/gravitino/catalog/lakehouse/paimon/integration/test/CatalogPaimonFileSystemIT.java [hunk 4]
```diff
@@ -48,4 +55,22 @@ protected Map<String, String> initPaimonCatalogProperties() {
 
     return catalogProperties;
   }
+
+  @Test
+  void testPaimonSchemaProperties() throws Catalog.DatabaseNotExistException {
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-paimon/src/test/java/org/apache/gravitino/catalog/lakehouse/paimon/integration/test/CatalogPaimonFileSystemIT.java:60]: no need to add this test for schema properties, since it's empty.

**Comment by FANNG1** [comment]
@caican00 , some minor comments, and could you add related document?

**Comment by caican00** [comment]
> @caican00 , some minor comments, and could you add related document?

ok.

File: catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java (hunk 1 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:49]: Paimon use `jdbc.` as property prefix.

File: catalogs/catalog-lakehouse-paimon/build.gradle.kts (hunk 3 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/build.gradle.kts:132]: > why adding this?

to test paimon catalog when the catalog backend is mysql.

File: catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/utils/CatalogUtils.java (hunk 2 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/utils/CatalogUtils.java]: fixed.

File: catalogs/catalog-lakehouse-paimon/src/test/java/org/apache/gravitino/catalog/lakehouse/paimon/integration/test/CatalogPaimonFileSystemIT.java (hunk 4 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/src/test/java/org/apache/gravitino/catalog/lakehouse/paimon/integration/test/CatalogPaimonFileSystemIT.java:60]: removed.

**Comment by caican00** [comment]
> @caican00 , some minor comments, and could you add related document?

@FANNG1 please review again, thanks

File: catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java (hunk 1 above)
Comment (FANNG1) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:52]: Iceberg internally uses `jdbc.user` too, we'd better provide a unified Gravitino Jdbc username.

File: catalogs/catalog-lakehouse-paimon/build.gradle.kts (hunk 3 above)
Comment (FANNG1) [reply catalogs/catalog-lakehouse-paimon/build.gradle.kts:133]: ok

File: catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java (hunk 1 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java]: updated. @FANNG1

File: catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java [hunk 5]
```diff
@@ -105,6 +119,18 @@ public class PaimonCatalogPropertiesMetadata extends BaseCatalogPropertiesMetada
                 "Paimon catalog uri config",
                 false /* immutable */,
                 null /* defaultValue */,
+                false /* hidden */),
+            stringOptionalPropertyEntry(
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:123]: Could you make `GRAVITINO_JDBC_USER` and `GRAVITINO_JDBC_PASSWORD` hidden?

Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:123]: updated.

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@caican00 , thanks for your contribute!
